modules/globebrowsing/util/globelabelscomponent.h

```diff
@@ -0,0 +1,133 @@
+* The above copyright notice and this permission notice shall be included in all copies *
+* or substantial portions of the Software.                                              *
+*                                                                                       *
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+* INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+* PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+* HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+* CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+* OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+****************************************************************************************/
+
+#ifndef __OPENSPACE_MODULE_GLOBEBROWSING___GLOBELABELSCOMPONENT___H__
+#define __OPENSPACE_MODULE_GLOBEBROWSING___GLOBELABELSCOMPONENT___H__
+
+#include <openspace/properties/propertyowner.h>
+
+#include <openspace/properties/scalar/boolproperty.h>
+#include <openspace/properties/scalar/floatproperty.h>
+#include <openspace/properties/scalar/intproperty.h>
+#include <openspace/properties/vector/vec4property.h>
+#include <openspace/properties/optionproperty.h>
+
+#include <ghoul/glm.h>
+#include <ghoul/font/fontrenderer.h>
+
+namespace ghoul { class Dictionary;}
+
+namespace ghoul::opengl { class ProgramObject; }
+
+namespace openspace {
+    struct RenderData;
+    
+    namespace documentation { struct Documentation; }
+    namespace globebrowsing { class RenderableGlobe; }
+
+    class GlobeLabelsComponent : public properties::PropertyOwner {
+    public:
+        enum labelRenderingAlignmentType {
+            Horizontally = 0,
+            Circularly = 1
+        };
+
+        // Labels Structures
+        struct LabelEntry {
+            char feature[256];
+            float diameter;
+            float latitude;
+            float longitude;
+            glm::vec3 geoPosition;
+        };
+        struct Labels {
+            std::string filename;
+            std::vector<LabelEntry> labelsArray;
+        };
+
+        const double labelFadeRangeConst = 1500.0;
+        const double rangeAngularCoefConst = 0.8;
+        const float minTransparencyValueConst = 0.009f;
+
+        GlobeLabelsComponent();
+        ~GlobeLabelsComponent() = default;
+
+        void initialize(
+            const ghoul::Dictionary& dictionary,
+            globebrowsing::RenderableGlobe* globe, 
+            std::shared_ptr<ghoul::fontrendering::Font> font = nullptr);
+        bool initializeGL();
+        void initializeFonts();
+        bool deinitialize();
+
+        bool isReady() const;
+
+        void update();
+
+        static documentation::Documentation Documentation();
+
+        void draw(const RenderData& data);
+
+    private:
```

Comment:
move these to the other `private` section; each section should only be ther eonce